src/test/nodesessionmetadata_test.cpp

```diff
@@ -210,3 +209,116 @@ TEST_F(NodeSessionMetadataTest, GetSubsessionSizeShouldThrowWhenNonExistingSubse
     auto subsessionMeta = meta.generateSubsessions("request", 5)[0];
     EXPECT_THROW(subsessionMeta.getSubsessionSize("nonExisting"), std::logic_error);
 }
+
+TEST_F(NodeSessionMetadataTest, GetShardIdNoSubsession) {
+    NodeSessionMetadata meta;
+    EXPECT_EQ(meta.getShardId(), 1);
```

Comment:
Do we enumerate shards starting from 1?

Reply:
There was a bug that without sharding we started from 1 and with sharding we started from 0. Fixed